src/components/Text.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Text as RNText, TextStyle, StyleSheet } from 'react-native';
+import { useBrandConfig } from 'src/hooks/useBrandConfig';
+import { colors } from '../theme';
+
+interface Props {
+  variant?: Variant;
+  style?: TextStyle;
+  children: React.ReactNode;
+}
+
+export function Text({ variant = 'base', style, children }: Props) {
+  const { styles } = useBrandConfig();
+
+  const mergedStyles = StyleSheet.flatten<TextStyle>([
+    defaultTextStyles.base,
+    styles.Text?.base,
+    styles.Text?.[variant],
+    style,
+  ]);
+
+  return <RNText style={mergedStyles}>{children}</RNText>;
+}
+
+export interface TextStyles {
+  base?: TextStyle;
+  body?: TextStyle;
+  heading?: TextStyle;
+  subHeading?: TextStyle;
+}
+
+type Variant = keyof TextStyles;
+
+export const defaultTextStyles: TextStyles = {
+  base: {
+    color: colors.text,
+  },
+};
```

Comment:
Something that might be worth prototyping, as an additive alternative to components using `useBrandConfig` like this, is an HOC that captures:
* useBrandConfig usage, but
* is injected with a mapping between expected brandConfig props & the component's props

In my mind, that'd allow some flexibility like:
- While some components might be coupled to `useBrandConfig`, all don't have to be.  i.e. There's a "component purity" factor here in my mind
- Using regular RN components (or some other component lib) instead of ours.  This particularly comes to mind for components that aren't unique to LifeOmic, like this `Text` example